src/nupic/algorithms/backtracking_tm.py

```diff
@@ -470,7 +470,7 @@ def write(self, proto):
     proto.pamCounter = self.pamCounter
     proto.collectSequenceStats = self.collectSequenceStats
     proto.resetCalled = self.resetCalled
-    proto.avgInputDensity = self.avgInputDensity
+    proto.avgInputDensity = self.avgInputDensity or -1.0
```

Comment:
Are you guarding against zero or None? It would be good to have a descriptive comment to help reader/maintainer